private/erlang_xrl_yrl.bzl

```diff
@@ -0,0 +1,89 @@
+load("//:erlang_app_info.bzl", "ErlangAppInfo")
+load("//:util.bzl", "path_join")
+load(":util.bzl", "erl_libs_contents")
+load(
+    "//tools:erlang_toolchain.bzl",
+    "erlang_dirs",
+    "maybe_install_erlang",
+)
+
+def unique_dirnames(files):
+    dirs = []
+    for f in files:
+        dirname = f.path if f.is_directory else f.dirname
+        if dirname not in dirs:
+            dirs.append(dirname)
+    return dirs
+
+def _dirname(path):
+    return path.rpartition("/")[0]
+
+def _impl(ctx):
+    package_dir = path_join(
+        ctx.label.workspace_root,
+        ctx.label.package,
+    )
+
+    (erlang_home, _, runfiles) = erlang_dirs(ctx)
+
+    outputs = ctx.outputs.outs
+
+    out_dirs = unique_dirnames(outputs)
+    if len(out_dirs) > 1:
+        fail(ctx.attr.outs, "do not share a common parent directory")
+    out_dir = out_dirs[0]
+
+    script = """set -euo pipefail
+
+TMP=$(mktemp -d || mktemp -d -t bazel-tmp)
+test -d "$TMP"
+trap "rm -fr '$TMP'" EXIT
```

Comment:
We don't need this when using `out_dir` directly.

Should we do a `mkdir -p {out_dir}` for consistency?